### PR TITLE
feat: Query Profile

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Review by MarkDuckworth — COMMENTED** [review]
Feedback for API reference documentation. This is not a release blocker, but ideally should be addressed before release.

You can preview the api reference docs generated for cloud.google.com using these two steps:
1) `yarn cloud-rad` to generate the _devsite folder
2) Install, configure, and serve the _devsite folder using instructions at https://www.npmjs.com/package/@google-cloud/cloud-rad

This is the best process I found to get my documentation right.

Also note that this PR will have missing content in the API reference at https://googleapis.dev/nodejs/firestore/latest. But that's using a different system, and the current target is CGC docs.

**Review by MarkDuckworth — APPROVED** [review]
Reviewed the remainder. LGTM, but the api reference comments should be considered before release.

File: dev/src/query-profile.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
```
[the remark this answers is not in the document]
Comment (ehsannas) [reply dev/src/query-profile.ts:2]: I had copied from aggregate.ts, which turns out to be the only file that has this tag :)
I'll remove it as it seems to be that it's not used in any license header in the repo.

**Review by MarkDuckworth — APPROVED** [review]
LGTM

**Pull request merged**